libkmod/libkmod-config.c

```diff
@@ -146,8 +146,9 @@ static int kmod_config_add_command(struct kmod_config *config, const char *modna
 	if (!l)
 		return -ENOMEM;
 
+	TAKE_PTR(cmd);
 	*list = l;
-	cmd = NULL;
```

Comment:
If we follow the approach used in `kmod_file_load_zlib()` aka `TAKE_PTR` on use, we'll get a memory leak in the above case. Since the annotations are off, this means clang-analyzer won't notice it, where only valgrind (in an error path) will flag it.

Generally I like the idea, although I am wondering if we cannot tweak the magic macro somehow, so we don't end up with issues like ^^

Reply:
Yeah, we can't do:

```
l = kmod_list_append(*list, TAKE_PTR(cmd));
```

and I didn't see a good alternative other than what I did here.